logstash/check.py

```diff
@@ -0,0 +1,304 @@
+        if url is None:
+            raise Exception("A URL must be specified in the instance")
+
+        # Support URLs that have a path in them from the config, for
+        # backwards-compatibility.
+        parsed = urlparse.urlparse(url)
+        if parsed[2] != "":
+            url = "%s://%s" % (parsed[0], parsed[1])
+        port = parsed.port
+        host = parsed.hostname
+
+        custom_tags = instance.get('tags', [])
+        service_check_tags = [
+            'host:%s' % host,
+            'port:%s' % port
+        ]
+        service_check_tags.extend(custom_tags)
+
+        # Tag by URL so we can differentiate the metrics
+        # from multiple instances
+        tags = ['url:%s' % url]
+        tags.extend(custom_tags)
+
+        timeout = instance.get('timeout') or self.DEFAULT_TIMEOUT
+
+        config = LogstashInstanceConfig(
+            service_check_tags=service_check_tags,
+            ssl_cert=instance.get('ssl_cert'),
+            ssl_key=instance.get('ssl_key'),
+            ssl_verify=instance.get('ssl_verify'),
+            tags=tags,
+            timeout=timeout,
+            url=url
+        )
+        return config
+
+    def _get_data(self, url, config, send_sc=True):
+        """ Hit a given URL and return the parsed json
+        """
+        auth = None
+
+        # Load SSL configuration, if available.
+        # ssl_verify can be a bool or a string (http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)
+        if isinstance(config.ssl_verify, bool) or isinstance(config.ssl_verify, str):
+            verify = config.ssl_verify
+        else:
+            verify = None
+        if config.ssl_cert and config.ssl_key:
+            cert = (config.ssl_cert, config.ssl_key)
+        elif config.ssl_cert:
+            cert = config.ssl_cert
+        else:
+            cert = None
+
+        try:
+            resp = requests.get(
+                url,
+                timeout=config.timeout,
+                headers=headers(self.agentConfig),
+                auth=auth,
+                verify=verify,
+                cert=cert
+            )
+            resp.raise_for_status()
+        except Exception as e:
+            if send_sc:
+                self.service_check(
+                    self.SERVICE_CHECK_CONNECT_NAME,
+                    AgentCheck.CRITICAL,
+                    message="Error {0} when hitting {1}".format(e, url),
+                    tags=config.service_check_tags
+                )
+            raise
+
+        return resp.json()
+
+    def _get_logstash_version(self, config):
+        """ Get the running version of logstash.
+        """
+        try:
+            data = self._get_data(config.url, config, send_sc=False)
+            version = data['version']
+            version = map(int, version.split('.')[0:3])
+        except Exception as e:
+            self.warning(
+                "Error while trying to get Logstash version "
```

Comment:
We could add the fact the integration is defaulting to `1.0.0` in the error message.